x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/test/integration/BootStrapTests.java

```diff
@@ -141,7 +141,6 @@ public void testLoadMalformedWatchRecord() throws Exception {
         });
     }
 
-    @AwaitsFix(bugUrl = "Supposedly fixed; https://github.com/elastic/x-pack-elasticsearch/issues/1915")
     public void testLoadExistingWatchesUponStartup() throws Exception {
```

Comment:
This test would fail approximately 1 in 50 times due to this bug.  
The final `assertBusy` would fail because the returned `getWatchesCount()` would be 1 less than expected (numWatches) due to the size of the internal map becoming corrupt.

If you iterated through the map and explicitly counted the number of items, it was correct, but the `size()` returned an incorrect value.